
--- python-babel-2.10.3.orig/tests/test_dates.py
+++ python-babel-2.10.3/tests/test_dates.py
@@ -726,14 +726,12 @@ def test_format_time():
                                tzinfo=timezone('America/New_York'), locale='en')
     assert custom == u"09 o'clock AM, Eastern Daylight Time"
 
-    t = time(15, 30)
-    paris = dates.format_time(t, format='full',
-                              tzinfo=timezone('Europe/Paris'), locale='fr_FR')
-    assert paris == u'15:30:00 heure normale d\u2019Europe centrale'
-    us_east = dates.format_time(t, format='full',
-                                tzinfo=timezone('America/New_York'), locale='en_US')
-    assert us_east == u'3:30:00 PM Eastern Standard Time'
-
+    with freezegun.freeze_time("2023-01-01"):
+        t = time(15, 30)
+        paris = dates.format_time(t, format='full', tzinfo=paris, locale='fr_FR')
+        assert paris == '15:30:00 heure normale d’Europe centrale'
+        us_east = dates.format_time(t, format='full', tzinfo=eastern, locale='en_US')
+        assert us_east == '3:30:00\u202fPM Eastern Standard Time'
 
 def test_format_skeleton():
     dt = datetime(2007, 4, 1, 15, 30)
